File: _week-2-solution.py
import rasterio
import numpy as np
import matplotlib.pyplot as plt
from rasterio.enums import Resampling

# File path
ortho_path = "data/Orthofoto_ZH/3612.tif"

# Open raster
with rasterio.open(ortho_path) as src:
    # Read all bands
    ortho = src.read()
    
    # Resample to lower resolution (aggregate by factor of 200)
    new_shape = (src.height // 200, src.width // 200)
    ortho_10 = src.read(out_shape=(src.count, *new_shape), resampling=Resampling.average)

# Rename bands (assuming original order is R, G, B, NIR)
R, G, B, NIR = ortho_10

# Display RGB composite (NIR as red, G as green, B as blue)
plt.figure(figsize=(8, 8))
plt.imshow(np.dstack((NIR, R, G)))
plt.title("RGB Composite (NIR, R, G)")
plt.axis("off")
plt.show()

# Compute NDVI
NDVI = (NIR - R) / (NIR + R)

# Plot NDVI
plt.figure(figsize=(8, 8))
plt.imshow(NDVI, cmap='RdYlGn')
plt.colorbar(label="NDVI")
plt.title("NDVI Map")
plt.axis("off")
plt.show()

File: _week-2-solution.R
library(terra)

ortho <- rast("data/Orthofoto_ZH/3612.tif")

ortho_10 <- terra::aggregate(ortho, 200)

names(ortho_10) <- c("R","G","B","NIR")


plotRGB(ortho_10, r = 4, g = 1, b = 2)


NDVI <- (ortho_10$NIR - ortho_10$R)/(ortho_10$NIR + ortho_10$R)

plot(NDVI)
